Step-by-step-Tutorials/Custom-Storage-Providers.md

```diff
@@ -167,9 +168,9 @@ The format decisions have already been made, so coding up the `WriteStateAsync`
 
 
 ``` csharp
-public async Task WriteStateAsync(string grainType, Orleans.GrainReference grainRef, IGrainState grainState)
+public async Task WriteStateAsync(string grainType, GrainReference grainRef, IGrainState grainState)
 {
-    Dictionary<string, object> dataValues = grainState.AsDictionary();
+    IDictionary<string, object> dataValues = ((GrainState)grainState.State).AsDictionary();
     JavaScriptSerializer serializer = new JavaScriptSerializer();
```

Comment:
we probably need to change this too, since if I'm not mistaken, we shouldn't be using `GrainState` any more (neither for reading nor writing). I'm not very familiar with the new approach, maybe @jason-bragg can weigh in